edgelet/iotedge/src/check/checks/aziot_edged_version.rs

```diff
@@ -164,16 +149,74 @@ impl AziotEdgedVersion {
             .get(1)
             .expect("unreachable: regex defines one capturing group")
             .as_str();
-        self.actual_version = Some(version.to_owned());
+        Ok(version.to_owned())
+    }
+
+    async fn inner_execute(&mut self, check: &mut Check) -> anyhow::Result<CheckResult> {
+        let actual_version = self.get_installed_version(check).await?;
+        let expected_version = if let Some(expected_aziot_edged_version) =
+            &check.expected_aziot_edged_version
+        {
+            expected_aziot_edged_version.clone()
+        } else {
+            if check.parent_hostname.is_some() {
+                // This is a nested Edge device so it may not be able to access aka.ms or github.com.
+                // In the best case the request would be blocked immediately, but in the worst case it may take a long time to time out.
+                // The user didn't provide the `expected_aziot_edged_version` param on the CLI, so we just ignore this check.
+                return Ok(CheckResult::Ignored);
+            }
+
+            // Nightly builds of debian packages have a '~' which isn't allowed in semver. We're only interested in
+            // parsing the major and minor versions anyway, so we can just truncate the version string.
+            let actual_semver = Version::parse(
+                actual_version
+                    .split(|c| c == '-' || c == '~')
+                    .next()
+                    .unwrap_or(&actual_version),
+            )
+            .context("could not parse actual version as semver")?;
```

Comment:
It maybe helpful for debugging to print out `actual_version` in the log here?

Reply:
Done